lib/main.dart

```diff
@@ -14,10 +14,10 @@ void main() async {
     MultiProvider(
       providers: [
         ChangeNotifierProvider<Simulations>(
-          builder: (context) => Simulations(),
+          create: (context) => Simulations(),
```

Comment:
this won't work with shared preferences after upgrading to provider 4.0.4. You should add another parameter lazy: false.
```dart
ChangeNotifierProvider<Simulations>(
          lazy: false,
          create: (context) => Simulations(),
)
```